include/matx/generators/meshgrid.h

```diff
@@ -31,89 +31,89 @@
 /////////////////////////////////////////////////////////////////////////////////
 
 #pragma once
-
+#include "matx/operators/permute.h"
 
 namespace matx
 {
+
+
   namespace detail {
-    template <typename T> class Meshgrid_X {
-      private:
-        std::array<T, 3> x_;
-        std::array<T, 3> y_;
 
-      public:
-        // dummy type to signal this is a matxop
-        using matxop = bool;
-        using scalar_type = T;
+    template <typename T1, int RANK, int AXIS> 
+      class MeshGridOp {
+        private:
+          T1 t1_;
+          std::array<index_t, RANK> shape_;
 
-        __MATX_INLINE__ std::string str() { return "meshgridx"; }	
+        public:
+          using matxop = bool;
+          using scalar_type = typename T1::scalar_type;
+          //typedef typename T1::scalar_type scalar_type;
 
-        Meshgrid_X(std::array<T, 3> x, std::array<T, 3> y) : x_(x), y_(y) {}
+          __MATX_INLINE__ std::string str() { return "meshgrid"; }
 
-        inline __MATX_DEVICE__ T operator()(index_t i, index_t j) const
-        {
-          return x_[0] + j * (x_[1] - x_[0]) / (x_[2] - 1);
-        }
+          __MATX_INLINE__ MeshGridOp(T1 t1, std::array<index_t, RANK> shape) : t1_(t1), shape_(shape) {
+            static_assert(shape.size() == RANK );
```

Comment:
Maybe add a second argument to each of these saying why the assert failed

Reply:
a user will never hit this assert since they use a different api.  it was more for matx developers.